pkg/query/storeset.go

```diff
@@ -331,16 +334,26 @@ func (s *StoreSet) getHealthyStores(ctx context.Context) map[string]*storeRef {
 }
 
 func externalLabelsFromStore(store *storeRef) string {
-	tsdbLabels := labels.Labels{}
-	for _, l := range store.labels {
-		tsdbLabels = append(tsdbLabels, labels.Label{
-			Name:  l.Name,
-			Value: l.Value,
-		})
+	tsdbLabelSetStrings := make([]string, 0, len(store.labelSets))
+	for _, ls := range store.labelSets {
+		tsdbLabels := labels.Labels{}
+		for _, l := range ls.Labels {
+			tsdbLabels = append(tsdbLabels, labels.Label{
+				Name:  l.Name,
+				Value: l.Value,
+			})
+		}
+		sort.Sort(tsdbLabels)
+		tsdbLabelSetStrings = append(tsdbLabelSetStrings, tsdbLabels.String())
+	}
+	sort.Strings(tsdbLabelSetStrings)
+
+	s := ""
```

Comment:
can we replace this with `return strings.Join(tsdbLabelSetStrings, ", )` or some other delimiter?

Reply:
or just "" delimiter to have exactly same thing as you have here.

Reply:
:man_facepalming: you're absolutely right

Reply:
Done.